project-base/storefront/urql/errorExchange.ts

```diff
@@ -23,46 +29,69 @@ export const getErrorExchange =
                         return;
                     }
 
+                    if (isClient && hasFriendlyUrlQueryFailedWith500(operation, error)) {
+                        if (publicRuntimeConfig.errorMessageVerbosity === 'devel') {
+                            throw new Error(JSON.stringify(mapGraphqlErrorForDevelopment(error.graphQLErrors[0])));
+                        }
+
+                        throw new Error('Internal Server Error');
+                    }
+
                     const isAuthError = error.response?.status === 401;
                     if (isAuthError) {
                         removeTokensFromCookies(context);
 
                         return;
                     }
 
-                    const parsedErrors = t ? getUserFriendlyErrors(error, t) : undefined;
-                    if (!parsedErrors) {
-                        return;
+                    if (publicRuntimeConfig.errorMessageVerbosity === 'devel') {
+                        handleErrorMessagesForDevelopment(error);
+                    } else {
+                        handleErrorMessagesForUsers(error, t, operation);
                     }
+                }),
+            );
+        };
+    };
 
-                    if (parsedErrors.userError) {
-                        logException(parsedErrors.userError);
-                    }
+const handleErrorMessagesForDevelopment = (error: CombinedError) => {
+    logException(error);
+    error.graphQLErrors
+        .map((graphqlError) => mapGraphqlErrorForDevelopment(graphqlError))
+        .forEach((simplifiedGraphqlError) => showErrorMessage(JSON.stringify(simplifiedGraphqlError)));
+};
 
-                    const isCartError = operation.query === CartQueryDocumentApi;
-                    if (isCartError) {
-                        handleCartError(parsedErrors);
+const handleErrorMessagesForUsers = (error: CombinedError, t: Translate | undefined, operation: Operation) => {
+    const parsedErrors = t ? getUserFriendlyErrors(error, t) : undefined;
+    if (!parsedErrors) {
+        return;
+    }
 
-                        return;
-                    }
+    if (parsedErrors.userError) {
+        logException(parsedErrors.userError);
+    }
 
-                    if (!parsedErrors.applicationError) {
-                        return;
-                    }
+    const isCartError = operation.query === CartQueryDocumentApi;
```

Comment:
According to my previous comment this won't probably be the case anyway, but, I understand idea of not having variable defined before it's used because function can return earlier, but I don't see any benefit here. It's just simple boolean (not some heavy function call) so it get's readability down for basically no benefits. Let's rather define it with other variables on top. What do you think?

Reply:
Im not sure if I understand you 100%, but if the idea is to have the variable defined on top, I can move it. I am more inclined towards defining the variable as close to where it is used, but in this particular case, it's a small difference.